tests/test_app.rs

```diff
@@ -182,16 +183,23 @@ fn test_simple_app() {
     }
 
     fn config(cfg: &mut web::ServiceConfig) {
+        let test_scope = web::scope("/test")
+            .service(web::resource("{id}").route(web::get().to(path_with_param_without_slash)));
+
+        // Non-boxed middlewares are only supported in actix 4, prevously use of the Compat middleware was required
+        #[cfg(feature = "actix4")]
+        let test_scope = test_scope
+            .wrap(Logger::default())
+            .wrap(DefaultHeaders::default().add(("X-Test", "Value")));
```

Comment:
would you mind creating a new test just for actix4, can be very simple, just testing this works, example: `test_non_boxed_bodies` ?